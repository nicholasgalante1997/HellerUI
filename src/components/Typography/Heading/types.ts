import { Properties } from 'csstype';
import { SafeReactFC } from '../../../utils';

export interface StyledHeaderProps {
    color?: string;
    scale?: 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800;
}

export interface NativeHeadingProps {
    as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
    customStyles?: Properties;
}

export type HeadingProps = SafeReactFC<NativeHeadingProps & StyledHeaderProps>;
