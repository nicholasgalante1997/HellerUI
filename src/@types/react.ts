import React from 'react';

export type SafeReactFC<T> = T & {
    children?: React.ReactNode | React.ReactNode[];
};
