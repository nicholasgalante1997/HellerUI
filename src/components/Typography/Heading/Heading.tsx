import React from 'react';
import StyledDict from './Dict';
import { HeadingProps } from './types';

const Heading = (props: HeadingProps) => {
  const {
    as = 'h2',
    children,
    color,
    customStyles = {},
    scale = 300,
    dataTestId = 'heading'
  } = props;
  const Component = StyledDict[as];
  return (
    <Component
      data-testid={dataTestId}
      style={customStyles}
      scale={scale}
      color={color}
    >
      {children}
    </Component>
  );
};

export default Heading;
