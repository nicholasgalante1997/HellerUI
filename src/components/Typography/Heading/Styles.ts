import styled from 'styled-components';
import { StyledHeaderProps } from './types';

export const H1 = styled.h1<StyledHeaderProps>`
    color: ${(props: StyledHeaderProps) => (props.color ? props.color : 'black')};
    font-family: 'Poppins Bold';
    font-size: 3.5rem;
    font-weight: ${(props) => props.scale};
    margin-block-start: 0px;
    margin-block-end: 0px;
`;

export const H2 = styled.h2<StyledHeaderProps>`
    color: ${(props: StyledHeaderProps) => (props.color ? props.color : 'black')};
    font-family: 'Poppins Bold';
    font-size: 3rem;
    font-weight: ${(props) => props.scale};
    margin-block-start: 0px;
    margin-block-end: 0px;
`;

export const H3 = styled.h3<StyledHeaderProps>`
    color: ${(props: StyledHeaderProps) => (props.color ? props.color : 'black')};
    font-family: 'Poppins Bold';
    font-size: 2.5rem;
    font-weight: ${(props) => props.scale};
    margin-block-start: 0px;
    margin-block-end: 0px;
`;

export const H4 = styled.h4<StyledHeaderProps>`
    color: ${(props: StyledHeaderProps) => (props.color ? props.color : 'black')};
    font-family: 'Poppins Bold';
    font-size: 2rem;
    font-weight: ${(props) => props.scale};
    margin-block-start: 0px;
    margin-block-end: 0px;
`;

export const H5 = styled.h5<StyledHeaderProps>`
    color: ${(props: StyledHeaderProps) => (props.color ? props.color : 'black')};
    font-family: 'Poppins Bold';
    font-size: 1.5rem;
    font-weight: ${(props) => props.scale};
    margin-block-start: 0px;
    margin-block-end: 0px;
`;

export const H6 = styled.h6<StyledHeaderProps>`
    color: ${(props: StyledHeaderProps) => (props.color ? props.color : 'black')};
    font-family: 'Poppins Bold';
    font-size: 1rem;
    font-weight: ${(props) => props.scale};
    margin-block-start: 0px;
    margin-block-end: 0px;
`;
