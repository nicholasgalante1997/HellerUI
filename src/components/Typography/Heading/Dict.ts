import * as StyledHeading from './Styles';

const StyledHeadingDict = {
  h1: StyledHeading.H1,
  h2: StyledHeading.H2,
  h3: StyledHeading.H3,
  h4: StyledHeading.H4,
  h5: StyledHeading.H5,
  h6: StyledHeading.H6
} as const;

export default StyledHeadingDict;
