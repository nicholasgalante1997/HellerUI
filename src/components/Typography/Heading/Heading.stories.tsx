/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';
import { Heading, HeadingProps, HeadingSizes } from './Heading';

export default {
  title: 'Heading',
  component: Heading,
  parameters: {
    componentSubtitle: 'A typography subset for section titles, page headings, etc.',
  },
} as Meta;

const Template: Story<HeadingProps> = (args) => (<Heading {...args}>Heller UI</Heading>);

export const Default = Template.bind({});
Default.args = {};

export const Sizes = (args: any) => (
  <div style={{ padding: 8 }}>
    <Heading asHeading={HeadingSizes.H1} {...args}>Heller UI</Heading>
    <Heading asHeading={HeadingSizes.H2} {...args}>Heller UI</Heading>
  </div>
);
