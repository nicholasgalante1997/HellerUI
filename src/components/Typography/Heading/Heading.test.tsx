/* eslint-disable no-restricted-syntax */
/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { render } from '@testing-library/react';
import Heading from './Heading';

const headingStates = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'] as const;
const headingScales = [100, 200, 300, 400, 500, 600, 700, 800] as const;

describe('Heading', () => {
  test('h1 - h6 snapshots', () => {
    for (const hState of headingStates) {
      const { container } = render(<Heading as={hState}>Heller UI</Heading>);
      expect(container.firstChild).toBeInTheDocument();
      expect(container.firstChild).toHaveStyle("font-family: 'Poppins Bold';");
      expect(container.firstChild).toMatchSnapshot();
    }
  });
  test('dynamic color', () => {
    const { container } = render(
      <Heading as="h3" color="deeppink">
        Heller UI
      </Heading>
    );
    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toHaveStyle('color: deeppink;');
    expect(container.firstChild).toMatchSnapshot();
  });
  test('font weight', () => {
    for (const scale of headingScales) {
      const { container } = render(
        <Heading as="h3" scale={scale}>
          Heller UI
        </Heading>
      );
      expect(container.firstChild).toBeInTheDocument();
      expect(container.firstChild).toHaveStyle(`font-weight: ${scale};`);
      expect(container.firstChild).toMatchSnapshot();
    }
  });
});
