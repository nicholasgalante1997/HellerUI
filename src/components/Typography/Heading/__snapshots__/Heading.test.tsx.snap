// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading dynamic color 1`] = `
<h3
  class="sc-dkzDqf crcEiQ"
  color="deeppink"
  data-testid="heading"
  scale="300"
>
  Heller UI
</h3>
`;

exports[`Heading font weight 1`] = `
<h3
  class="sc-dkzDqf EcPsp"
  data-testid="heading"
  scale="100"
>
  Heller UI
</h3>
`;

exports[`Heading font weight 2`] = `
<h3
  class="sc-dkzDqf cDVHfq"
  data-testid="heading"
  scale="200"
>
  Heller UI
</h3>
`;

exports[`Heading font weight 3`] = `
<h3
  class="sc-dkzDqf dOTpKf"
  data-testid="heading"
  scale="300"
>
  Heller UI
</h3>
`;

exports[`Heading font weight 4`] = `
<h3
  class="sc-dkzDqf jNBFDo"
  data-testid="heading"
  scale="400"
>
  Heller UI
</h3>
`;

exports[`Heading font weight 5`] = `
<h3
  class="sc-dkzDqf kYzoid"
  data-testid="heading"
  scale="500"
>
  Heller UI
</h3>
`;

exports[`Heading font weight 6`] = `
<h3
  class="sc-dkzDqf bIWwwi"
  data-testid="heading"
  scale="600"
>
  Heller UI
</h3>
`;

exports[`Heading font weight 7`] = `
<h3
  class="sc-dkzDqf cTUfaX"
  data-testid="heading"
  scale="700"
>
  Heller UI
</h3>
`;

exports[`Heading font weight 8`] = `
<h3
  class="sc-dkzDqf hbuLRI"
  data-testid="heading"
  scale="800"
>
  Heller UI
</h3>
`;

exports[`Heading h1 - h6 snapshots 1`] = `
<h1
  class="sc-bczRLJ ihptbP"
  data-testid="heading"
  scale="300"
>
  Heller UI
</h1>
`;

exports[`Heading h1 - h6 snapshots 2`] = `
<h2
  class="sc-gsnTZi cynywS"
  data-testid="heading"
  scale="300"
>
  Heller UI
</h2>
`;

exports[`Heading h1 - h6 snapshots 3`] = `
<h3
  class="sc-dkzDqf dOTpKf"
  data-testid="heading"
  scale="300"
>
  Heller UI
</h3>
`;

exports[`Heading h1 - h6 snapshots 4`] = `
<h4
  class="sc-hKMtZM inZEFP"
  data-testid="heading"
  scale="300"
>
  Heller UI
</h4>
`;

exports[`Heading h1 - h6 snapshots 5`] = `
<h5
  class="sc-eCYdqJ hXDmpu"
  data-testid="heading"
  scale="300"
>
  Heller UI
</h5>
`;

exports[`Heading h1 - h6 snapshots 6`] = `
<h6
  class="sc-jSMfEi TLEWf"
  data-testid="heading"
  scale="300"
>
  Heller UI
</h6>
`;
